src/main/java/frc/robot/Apriltags.java

```diff
@@ -0,0 +1,164 @@
+               double height = kTargetHeight - kLensHeight;
+               double angleInRadians = (kMountAngle + veritcalTargetAngle) * (Math.PI / 180);
+
+               double horizontalDistance = height / Math.tan(angleInRadians);
+               //Determines the distance with camera height minus target height devided by tangent of camera angle plus the vertical offset angle
+               return horizontalDistance;
+           }
+       
+           public double estimateVerticalDistance(double horizontalDistance){
+               double horizontalTargetAngle = getHorizontalDegToTarget();
+               
+               double verticalDistance = horizontalDistance * Math.tan(horizontalTargetAngle);
+               //Determines the distance with vertical distance multiply by the tangent of horizontal target angle
+               return verticalDistance;
+           }
+       
+           public boolean isDistancePossible(){
+                drive.angleAlign();
+                boolean distPossible = false;
+               if (getPipeline() == 0) {
+                    distPossible = ((estimateHorizontalDistance() < maxArmReach && estimateHorizontalDistance() > minArmReach) ? true : false);
+           }
+               if (getPipeline() == 1) {
+                    distPossible = (((estimateHorizontalDistance() + 17.375)  < maxArmReach && (estimateHorizontalDistance() + 17.375) > minArmReach) ? true : false);
+               }
+               return distPossible;
+           }
+
+           public boolean isScoringPossible(){
+            return((estimateHorizontalDistance() < 28.6) ? true : false);
+           }
+        
+           //Determines if the distance to target is posible
+
+        //    public double alignLongitude() {
+        //     setPipeline(1);
+        //     double distance = 0;
+        //     if (getHorizontalDegToTarget() < -1 || getHorizontalDegToTarget() > 1) {
+        //         if (getPipeline() == 1) {
+        //             drive.angleAlign();
+        //         }
+        //     } 
+        //     if (getHorizontalDegToTarget() < 1 && getHorizontalDegToTarget() > -1) {
+        //         double length1 = estimateVerticalDistance();
+        //         double offset1 = getHorizontalDegToTarget();
+
+        //         // drive.leftFront.stopMotor();
+        //         // drive.rightFront.stopMotor();
+        //         // drive.leftBack.stopMotor();
+        //         // drive.rightBack.stopMotor();
+        //         setPipeline(2);
+        //         double length2 = estimateVerticalDistance();
+        //         double offset2 = getHorizontalDegToTarget();
+
+        //         double theta = (offset2 - offset1) * (Math.PI/180);
+        //         double angle1 = Math.asin(length2 * (Math.sin(theta)/17));
+    
+        //         double angle2 = 180 - 90 - angle1;
+        //         distance = length1 * Math.tan(angle2);
+        //     }
+        //     return distance;
+        // }
```

Comment:
I'd recommend adding in a comment about the purpose of this method or if it's still needed for final robot code... if not, you can delete it here since it'll still be stored in git hub history for reference